tests/test_BinnedSpectrum.py

```diff
@@ -0,0 +1,29 @@
+from ms2deepscore import BinnedSpectrum
+
+
+def test_BinnedSpectrum():
+    """basic test for BinnedSpectrum class"""
+    peaks = {'20': 0.05,
+             '21': 0.1,
+             '25': 0.5,
+             '30': 0.5,
+             '100': 1.0}
+    metadata = {"inchikey": "test"}
+    binned_spectrum = BinnedSpectrum(binned_peaks=peaks,
+                                     metadata=metadata)
+    assert binned_spectrum.peaks["20"] == 0.05
+    assert list(binned_spectrum.peaks.keys()) == ['20', '21', '25', '30', '100']
+    assert binned_spectrum.get("inchikey") == "test"
+    assert binned_spectrum.get("smiles") is None
+
+
+def test_BinnedSpectrum_deepcopy():
+    """Test if metadata is made a correct deepcopy"""
```

Comment:
You are doing a shallow copy in the code I think. I don't know a lot about this, but I suspect that if instead of a string you try mutating a more complex object that this test fails.

Reply:
I thought I am doing a deepcopy already. Can you think of an example that should fail?